### ISSUE TITLE
(sphinx) Prevent search from returning deprecated methods and/or prevent deprecated methods from being documented

### ISSUE THREAD

**Issue by deanm0000** [pola-rs/polars#14286]
### Description

Say I search for horizontal

![image](https://github.com/pola-rs/polars/assets/37878412/5bb502aa-9909-4957-adc2-3e0a5837e075)

then we don't want the deprecated methods to be returned.

I'm not a sphinx user but [this](https://stackoverflow.com/questions/8313476/how-can-i-configure-sphinx-to-conditionally-exclude-some-pages) seems relevant.

I'm not sure if there's a way to make it only prevent them from showing up in search results or if it has to be excluded entirely from the docs.

**Comment by mcrumiller** [comment]
I disagree, if a user's code uses a function that works, we should have documentation for it, even if it's deprecated. What if they hand off the code to someone else, and they need to find out what that function does?

It would be nice on the other hand if the search results had a `deprecated` label but that's probably not possible.
